src/main/resources/swagger/api-docs.yaml

```diff
@@ -1044,6 +1044,138 @@ paths:
           description: Success
         500:
           description: Internal Error
+  /profile:
+    get:
+      tags:
+        - Profile
+      operationId: getAll
+      summary: Returns a list of all keys and values stored in the user profile service for the currently logged-in user.
+      parameters:
+      responses:
+        200:
+          description: OK
+          produces:
+          - application/json
+          schema:
+            {
+              "properties": {
+                "userId": {
+                  type: "string"
+                },
+                "keyValuePairs": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "required": [ "key", "value" ],
+                    "properties": {
+                      "key": {
+                        "type": "string"
+                      },
+                      "value": {
+                        "type": "string"
+                      }
+                    }
+                  }
+                }
+              },
+              "type": "object",
+              "required": [ "userId", "keyValuePairs" ]
+            }
+        500:
+          description: Internal server error
+          produces: text/plain
+  /profile/{key}:
+    post:
+      tags:
+        - Profile
+      operationId: setKey
+      summary: Adds or resets a single key/value pair in the user profile service for the currently logged-in user.
+      consumes:
+      - text/plain
```

Comment:
This may be the only place we POST a text/plain value instead of a json object. Should I change this to json?